java-extensions/jdbc-bridge/src/main/java/com/starrocks/jdbcbridge/JDBCScanner.java

```diff
@@ -13,28 +13,38 @@
 import java.util.List;
 
 public class JDBCScanner {
-    private JDBCScanContext scanContext;
+    private String driverLocation;
     private HikariDataSource dataSource;
+    private JDBCScanContext scanContext;
     private Connection connection;
     private Statement statement;
     private ResultSet resultSet;
     private ResultSetMetaData resultSetMetaData;
     private List<String> resultColumnClassNames;
 
-    public JDBCScanner(JDBCScanContext scanContext) {
+    public JDBCScanner(String driverLocation, JDBCScanContext scanContext) {
+        this.driverLocation = driverLocation;
         this.scanContext = scanContext;
     }
 
     public void open() throws Exception {
-        HikariConfig config = new HikariConfig();
-        config.setDriverClassName(scanContext.getDriverClassName());
-        config.setJdbcUrl(scanContext.getJdbcURL());
-        config.setUsername(scanContext.getUser());
-        config.setPassword(scanContext.getPassword());
-        // one connection per query, so just set max pool size to 1
-        config.setMaximumPoolSize(1);
+        final String user = scanContext.getUser();
+        final String pass = scanContext.getPassword();
+        final String url = scanContext.getJdbcURL();
+
+        dataSource = DataSourceCache.getInstance().getSource(driverLocation, user, pass, url);
```

Comment:
maybe we need a lock here or getOrCreate interface?

Reply:
I think it's unnecessary